dpc/src/account/private_key.rs

```diff
@@ -72,12 +79,20 @@ impl<C: Parameters> PrivateKey<C> {
 impl<C: Parameters> From<&C::AccountSeed> for PrivateKey<C> {
     /// Returns the account private key from an account seed.
     fn from(seed: &C::AccountSeed) -> Self {
+        // Construct the sk_sig domain separator.
+        let sk_sig_input = ACCOUNT_SEED_SK_SIG_DOMAIN;
+        let sk_sig_domain = C::ProgramScalarField::from_bytes_le_mod_order(&sk_sig_input.as_bytes());
+
+        // Construct the r_sig domain separator.
+        let r_sig_input = format!("{}_{}", ACCOUNT_SEED_R_SIG_DOMAIN, 0);
+        let r_sig_domain = C::ProgramScalarField::from_bytes_le_mod_order(r_sig_input.as_bytes());
```

Comment:
Maybe we should hash these using Blake2s first? If we're not checking this derivation in circuit (and we probably shouldn't be), then hashing makes sure it fits in 32 bytes.